src/chain/tezos/contracts/HicNFT.ts

```diff
@@ -0,0 +1,232 @@
+        allTokens: number;
+        ledgerMapId: number;
+        metadataMapId: number;
+        operatorsMapId: number;
+        paused: boolean;
+        tokenMetadataMapId: number;
+    }
+
+    export const objktsAddress = 'KT1RJ6PbjHpwc3M5rw5s2Nbmefwbuwbdxton';
+    export const hDaoAddress = 'KT1AFA2mwNUMNd4SsujE1YYp29vd8BZejyKW';
+
+    /*
+     * Objkts FA2 contract storage type
+     */
+    export type ObjktsStorage = HicStorage;
+
+    /*
+     * Get an instance of a Objkts contract's storage by querying a given address
+     * @param server The Tezos node to communicate with
+     * @param address The deployed Timelock contract address
+     */
+    export async function getObjktsStorage(server: string): Promise<ObjktsStorage> {
+        const storageResult = await TezosNodeReader.getContractStorage(server, objktsAddress);
+        return {
+            administrator: JSONPath({path: '$.args[0].args[0].string', json: storageResult })[0],
+            allTokens: Number(JSONPath({ path: '$.args[0].args[1].int', json: storageResult })[0]),
+            ledgerMapId: Number(JSONPath({ path: '$.args[0].args[2].int', json: storageResult })[0]),
+            metadataMapId: Number(JSONPath({ path: '$.args[1].args[0].int', json: storageResult })[0]),
+            operatorsMapId: Number(JSONPath({ path: '$.args[1].args[1].int', json: storageResult })[0]),
+            paused: (JSONPath({ path: '$.args[2].prim', json: storageResult })[0]).toString().toLowerCase().startsWith('t'),
+            tokenMetadataMapId: Number(JSONPath({ path: '$.args[3].int', json: storageResult })[0])
+        };
+    }
+
+    /*
+     * hDao FA2 contract storage type
+     */
+    export type HDaoStorage = HicStorage;
+
+
+    /*
+     * Get an instance of an hDao contract's storage by querying a given address
+     * @param server The Tezos node to communicate with
+     * @param address The deployed Timelock contract address
+     */
+    export async function getHDaoStorage(server: string): Promise<HDaoStorage> {
+        const storageResult = await TezosNodeReader.getContractStorage(server, hDaoAddress);
+        return {
+            administrator: JSONPath({path: '$.args[0].args[0].string', json: storageResult })[0],
+            allTokens: Number(JSONPath({ path: '$.args[0].args[1].int', json: storageResult })[0]),
+            ledgerMapId: Number(JSONPath({ path: '$.args[0].args[2].int', json: storageResult })[0]),
+            metadataMapId: Number(JSONPath({ path: '$.args[1].args[0].int', json: storageResult })[0]),
+            operatorsMapId: Number(JSONPath({ path: '$.args[1].args[1].int', json: storageResult })[0]),
+            paused: (JSONPath({ path: '$.args[2].prim', json: storageResult })[0]).toString().toLowerCase().startsWith('t'),
+            tokenMetadataMapId: Number(JSONPath({ path: '$.args[3].int', json: storageResult })[0])
+        };
+    }
+
+    /*
+     * FA2 individual token transfer
+     */
+    interface TokenTransaction {
```

Comment:
@itamarreif: instead of defining new types, use what's available in [MultiAssetTokenHelper](https://github.com/Cryptonomic/ConseilJS/blob/master/src/chain/tezos/contracts/tzip12/MultiAssetTokenHelper.ts).